tests/test_wheel.py

```diff
@@ -45,3 +52,62 @@ def test_find_metadata_files():
     ]
     candidates = wheel.Wheel.find_candidate_metadata_files(names)
     assert expected == candidates
+
+
+def test_read_valid(example_wheel):
+    """Test reading a valid wheel file"""
+    metadata = example_wheel.read().decode().splitlines()
+    assert "Name: twine" in metadata
+    assert "Version: 1.5.0" in metadata
+
+
+def test_read_non_existent_wheel_file_name():
+    """Test reading a wheel file which doesn't exist"""
+
+    file_name = "/foo/bar/baz.whl"
+    with pytest.raises(exceptions.InvalidDistribution) as excinfo:
```

Comment:
You could avoid the following `assert` by using the `match` kwarg to `pytest.raises`:

```suggestion
    with pytest.raises(exceptions.InvalidDistribution, match="No such file: " + file_name) as excinfo:
```

Reply:
Aah, didn't know of this syntax, will change it.

Reply:
Updated the syntax as per suggestion, with f-strings used for formatting